toolkit/smart-contracts/offchain/src/governance.rs

```diff
@@ -252,6 +252,41 @@ pub async fn get_governance_policy_summary<T: QueryLedgerState + QueryNetwork>(
 	}
 }
 
+#[derive(Clone, PartialEq, Eq, Hash)]
+pub struct MultiSigParameters {
+	governance_authorties: Vec<MainchainKeyHash>,
+	threshold: u8,
+}
+
+impl MultiSigParameters {
+	pub fn new(governance_authorties: &[MainchainKeyHash], threshold: u8) -> Result<Self, &str> {
+		if governance_authorties.is_empty() {
+			return Err("governance authorities cannot be be empty");
+		}
+		if threshold == 0 {
+			return Err("threshold has to be a positive number");
+		}
+		if threshold as usize > governance_authorties.len() {
```

Comment:
try to avoid `as`, consider:
```suggestion
		if usize::from(threshold) > governance_authorties.len() {
```

Reply:
Done, thanks